### DISCUSSION TITLE
[Questions] Horizontal server scaling behind a load balancer

### DISCUSSION THREAD

**Discussion by gnito-org** [mastodon/mastodon#18153] [Q&A]
Assuming that there is a central AWS S3 bucket for file storage, a central AWS RDS PostgreSQL server and a central AWS Redis server for a Mastodon installation:

1) Is Mastodon architected so that I can run five identical EC2 server copies of the Mastodon installation behind an AWS ELB load balancer that sends web and API requests to the EC2 servers on a round-robin basis?

2) If yes to 1), will the background processes that run on all five servers avoid task duplication?

3) If yes to 1), can the number of EC2 servers be varied via auto-scaling without any changes to the Mastodon configuration?

3) Does the Mastodon Redis process use a domain name so that the Mastodon data does not clash with any other applications that might also use the central Redis server?

**Comment by marians** [comment]
To answer one of the questions:

> 2\. If yes to 1), will the background processes that run on all five servers avoid task duplication?

According to https://docs.joinmastodon.org/admin/scaling/#sidekiq-queues, one of the task queues is not safe to run in this setup.
